src/scienceDatabase.cpp

```diff
@@ -14,30 +14,36 @@ REGISTER_SCRIPT_CLASS(ScienceDatabase)
     REGISTER_SCRIPT_CLASS_FUNCTION(ScienceDatabase, setImage);
 }
 
-
+unsigned int ScienceDatabase::next_id = 1;
```

Comment:
No need to keep an ID count in the science database, each multiplayer object already gets an unique ID:
https://github.com/daid/SeriousProton/blob/master/src/multiplayer.h#L231